tsdb/head.go

```diff
@@ -1177,7 +1202,7 @@ func (h *Head) gc() {
 
 	h.symbols = symbols
 	h.values = values
-
+	h.metrics.collectLabelMetrics(h.values)
```

Comment:
Iterating over ever every label name each time a series is created seems like it may cause performance problems.